File: src/main/java/application/controller/StudentController.java
package application.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import application.model.Student;
import application.model.Subject;
import application.service.StudentService;

@RestController
@RequestMapping("/students")
public class StudentController {

	@Autowired
	private StudentService studService;

	@GetMapping(value = "/{fileNumber}")
	public Student getStudentByFileNumber(@PathVariable Integer fileNumber) {
		return studService.getStudentByFileNumber(1);
	}

	@PostMapping(value = "/subscribe/{fileNumber}", consumes = "application/json")
	public void subscribe( @RequestBody Subject subject, @PathVariable Integer fileNumber){
		Student st = studService.getStudentByFileNumber(fileNumber);
		st.getSubjects().stream().filter(s -> s.getName().equals(subject.getName())).findFirst().get().setSubscribed(true);
		studService.save(st);
	}

	@PostMapping(value = "/unsubscribe/{fileNumber}", consumes = "application/json")
	public void unsubscribe( @RequestBody Subject subject, @PathVariable Integer fileNumber){
		Student st = studService.getStudentByFileNumber(fileNumber);
		st.getSubjects().stream().filter(s -> s.getName().equals(subject.getName())).findFirst().get().setSubscribed(false);
		studService.save(st);
	}
}

File: src/main/kotlin/ar/edu/unq/eiroalando/controller/StudentController.kt
package ar.edu.unq.eiroalando.controller

import ar.edu.unq.eiroalando.model.Subject
import ar.edu.unq.eiroalando.model.User
import ar.edu.unq.eiroalando.service.UserService
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.beans.factory.annotation.Autowired



@RestController
@RequestMapping("/students")
class StudentController {

    @Autowired
    lateinit var studService: UserService

    @GetMapping(value = "/{fileNumber}")
    fun getStudentByFileNumber(@PathVariable fileNumber: Int): User {
        return studService!!.getStudentByFileNumber(fileNumber)
    }

    @GetMapping(value = "/all")
    fun getAllStudents(): List<User>? {
        return studService.findStudents()?.toList()
    }

    @PostMapping(value = "/subscribe/{fileNumber}", consumes = arrayOf("application/json"))
    fun subscribe(@RequestBody subject: Subject, @PathVariable fileNumber: Int?) {
        val st = studService!!.getStudentByFileNumber(fileNumber!!)
        studService!!.save(st)
    }

    @PostMapping(value = "/add", consumes = arrayOf("application/json"))
    fun register(@RequestBody student: User) {
        studService.save(student)
    }

    @PostMapping(value = "/update", consumes = arrayOf("application/json"))
    fun update(@RequestBody student: User) {
        studService.update(student)
    }

    @PostMapping(value = "/unsubscribe/{fileNumber}", consumes = arrayOf("application/json"))
    fun unsubscribe(@RequestBody subject: Subject, @PathVariable fileNumber: Int?) {
        val st = studService!!.getStudentByFileNumber(fileNumber!!)
        studService!!.save(st)
    }

}